adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -253,7 +253,8 @@ private void updatePreferredNetworkLocation(final URL authorityUrl, final Authen
                 final URL replacedAuthority = Utility.constructAuthorityUrl(authorityUrl, metadata.getPreferredNetwork());
                 request.setAuthority(replacedAuthority.toString());
             } catch (final MalformedURLException ex) {
-                throw new AuthenticationException(ADALError.DEVELOPER_AUTHORITY_IS_NOT_VALID_URL, ex.getMessage(), ex);
```

Comment:
From my testing, a null preferred network doesn't actually result in a MalformedURLException. The resulting URL just has the empty string as its authority. There should be an explicit null check around the preferred network.

Reply:
It could not be null, it the metadata is validated, the preferred network won't be null. If the metadata has not been validated, the performAuthorityValidation will return before reaching the updatePreferredNetworkLocation code. The MalformedURLException will be thrown if the authority is not a valid url.

Reply:
@rygo-msft is correct. if authority is null, we will create an authority url as https:///common. Lets add a null check as well

Reply:
updated per comments.

Reply:
According to the change in Discovery, the preferredNetwork won't be null when isValidated==true. And AcquireTokenRequest.java L251 won't be reached unless isValidated==true. I'm OK to add the null check on the getPreferredNetwork. But it seems redundant to me.

Reply:
Based on how the constructor for InstanceDiscoveryMetadata(bool) is written, isValidate can be true with preferred network being false. It may be worth refactoring that constructor since InstanceDiscoveryMetadata(false) is the only valid way to call it.